src/st/no3/377D4.c

```diff
@@ -412,9 +419,27 @@ INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C4D4C);
 
 INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C4E4C);
 
-INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C4F64);
+s16 func_801C4F64(void) {
+    s16 temp_v1;
+    s16 var_a0;
+
+    temp_v1 = (u16) D_8006C3B8->posX.Data.high - (u16) D_800733DA;
+    var_a0 = temp_v1;
+    if (temp_v1 >> 0x10) {
+        var_a0 = -temp_v1;
+    }
+    return var_a0;
+}
```

Comment:
```suggestion
    s16 temp_v1;

    temp_v1 = (u16) D_8006C3B8->posX.Data.high - (u16) D_800733DA;
    var_a0 = temp_v1;
    if (temp_v1 >> 0x10) {
        temp_v1 = -temp_v1;
    }
    return temp_v1;
}
```
The temp might not be necessary, does this match?

Reply:
one variable was unecessary indeed !
this matched:

s16 func_801C4F64(void) {
    s16 temp_v1;

    temp_v1 = D_8006C3B8->posX.Data.high - D_800733DA;
    if (temp_v1 < 0) {
        temp_v1 = -temp_v1;
    }
    return temp_v1;
}